### ISSUE TITLE
Typescript with eslint: 'property' is missing in props validation eslint(react/prop-types)

### ISSUE THREAD

**Issue by aleczratiu** [eslint/eslint#13284]
We are using typescript in our project and when I try to access some property from an object, occur an eslint error as in the example below:

> 'color.left' is missing in props validation eslint(react/prop-types)

**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **React Version:** 16.13.1
* **typescript-eslint/eslint-plugin** 2.21.0
* **typescript-eslint/parser** 2.21.0
* **eslint-plugin-react**: 7.19.0

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

import React, { FunctionComponent } from 'react';

type TestCompColor = {
    left: number;
    top: number;
}

type TestCompProps = {
    value: string;
    color: TestCompColor;
}

const TestComp: FunctionComponent<TestCompProps> = ({ value, color }: TestCompProps) => {
    const getColor = () => {
        return (
            <div>
                {value}
                {color.left}
            </div>
        );
    };

    return (
        <div>
            {value}
            {getColor()}
        </div>
    );
};

export default TestComp;

```

**What actually happened? Please include the actual, raw output from ESLint.**

'color.left' is missing in props validation eslint(react/prop-types)

<img width="684" alt="Screenshot 2020-05-12 at 08 50 25" src="https://user-images.githubusercontent.com/33958032/81643137-a459a900-942d-11ea-9b75-9d4a58340d8c.png">

**Comment by anikethsaha** [comment]
Thanks for the issue. 

I think it is because `eslint-plugin-react/prop-types`  doesn't parse/uses typescript's types.  
IMO you can turn off this rule as you are using typescript which does that for you. or you can write proptypes for the component as well.

*not sure though, I might miss something*

**Comment by aleczratiu** [comment]
@anikethsaha yeah, but we also have a lot of old written component js/jsx where should work with `prop-types` check because only recently we migrate to typescript.

Also I still get this error just in that function scope, if I moved that destructuring `color.left` outside of that scope all seems to be working well, very strange.

**Comment by anikethsaha** [comment]
the code you have given seems fine, the rule is reporting because of no proptypes mentioned. 
IMO, you can turn off the rule as I mentioned above cause of TS. 



> Also I still get this error just in that function scope, if I moved that destructuring `color.left` outside of that scope all seems to be working well, very strange.

This might be because proptypes are for components, outside of function scope is non react component space, so no error from the rule. 

even if this is an issue (which I don't think it is), you can open an issue in [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react) as it is not related to eslint's core rules.

let me know if i missed something

**Comment by kaicataldo** [comment]
Closing this because we don't maintain the package in question.
